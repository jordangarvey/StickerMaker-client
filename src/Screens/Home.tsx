import React, { PureComponent } from "react";
import { StyleSheet } from "react-native";
import { NavigationScreenProp } from "react-navigation";

import ScreenContainer from "./ScreenContainer";
import Button from "../Components/Controls/Button";


/**
 * Component to render the Home screen
 */
class Home extends PureComponent<NavigationScreenProp<null>> {
	public constructor(props: any) {
		super(props);

		this.navigateToTemplatePicker = this.navigateToTemplatePicker.bind(this);
	}

	private navigateToTemplatePicker() {
		this.props.navigation.navigate("TemplatePicker");
	}

	/**
	 * Render the Home screen
	 */
	public render() {
		return (
			<ScreenContainer>
				<Button onPress={this.navigateToTemplatePicker}>Create a sticker</Button>
			</ScreenContainer>
		);
	}
}

/**
 * Styles for the Home screen
 */
const styles = StyleSheet.create({

});

export default Home;
