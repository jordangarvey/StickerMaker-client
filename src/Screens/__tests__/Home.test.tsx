import React from "react";
import { Text } from "react-native";
import renderer from "react-test-renderer";

import Home from "../Home";


describe("The Home componentâ€™s snapshot", () => {
	it("renders as expected", () => {
		const component = renderer.create(
			<Home/>
		).toJSON();


		expect(component).toMatchSnapshot();
	});
});
