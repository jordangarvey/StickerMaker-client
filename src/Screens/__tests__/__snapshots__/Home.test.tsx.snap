// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Home componentâ€™s snapshot renders as expected 1`] = `
<RCTScrollView
  alwaysBounceVertical={false}
  indicatorStyle="black"
  keyboardDismissMode="interactive"
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "bottom": 0,
      "left": 0,
      "paddingHorizontal": 40,
      "paddingTop": 50,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 12,
          "borderWidth": 1,
          "color": "black",
          "fontSize": 24,
          "opacity": 1,
          "overflow": "hidden",
          "padding": 12,
          "textAlign": "center",
        }
      }
    >
      <Text>
        Create a sticker
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
